signalfx/resource_signalfx_metric_ruleset.go

```diff
@@ -115,6 +116,85 @@ func metricRulesetResource() *schema.Resource {
 					},
 				},
 			},
+			"exception_rules": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Exception rules in the ruleset",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Name of this exception rule",
+						},
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "Status of this exception rule",
+						},
+						"matcher": {
+							Type:        schema.TypeSet,
+							Required:    true,
+							Description: "The matcher for this rule",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:         schema.TypeString,
+										Required:     true,
+										Description:  "The type of the matcher",
+										ValidateFunc: validation.StringInSlice([]string{"dimension"}, false),
+									},
+									"filters": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: "List of filters to match on",
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"not": {
+													Type:        schema.TypeBool,
+													Required:    true,
+													Description: "Flag specifying equals or not equals",
+												},
+												"property": {
+													Type:        schema.TypeString,
+													Required:    true,
+													Description: "Name of dimension to match",
+												},
+												"property_value": {
+													Type:        schema.TypeSet,
+													Required:    true,
+													Description: "List of dimension values to match",
+													Elem:        &schema.Schema{Type: schema.TypeString},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"restoration": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							Description: "Restoration for this rule",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"restoration_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+										// Optional:    true,
+										Description: "ID of the restoration job.",
+									},
+									"start_time": {
+										Type:        schema.TypeString,
```

Comment:
Any reason why not keep this a `long` like we do in the API?

Reply:
There is no schema.TypeLong.  Alternates are TypeInt and TypeFloat.
See https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-types

Also, other parts of the payload like last_updated is TypeString.  Let me see what happens if I change them all to TypeInt.